File: java/io/stpettersen/blackjack/IActor.java
/**
 * Blackjack
 * Copyright 2015 Sam Saint-Pettersen
 * Released under the MIT/X11 License.
 * 
 * Java Swing/AWT implementation.
*/

package io.stpettersen.blackjack;

public interface IActor
{
    int calcTotal();
    boolean hasBlackjack();
    boolean isBust();
    int showCards();
}

File: scala/src/io/stpettersen/blackjack/IActor.scala
/**
 * Blackjack
 * Copyright 2015 Sam Saint-Pettersen
 * Released under the MIT/X11 License.
 *
 * Scala Swing/AWT implementation.
 */

package io.stpettersen.blackjack

trait IActor {
  def calcTotal: Int
  def hasBlackjack: Boolean
  def isBust: Boolean
  def showCards: Int
}
